extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/QuarkusExecutorFactory.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.vertx.core.runtime;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadFactory;
+
+import org.jboss.threads.EnhancedQueueExecutor;
+import org.jboss.threads.JBossExecutors;
+import org.wildfly.common.cpu.ProcessorInfo;
+
+import io.quarkus.vertx.core.runtime.config.VertxConfiguration;
+import io.vertx.core.spi.ExecutorServiceFactory;
+
+public class QuarkusExecutorFactory implements ExecutorServiceFactory {
+    private final VertxConfiguration conf;
+
+    public QuarkusExecutorFactory(VertxConfiguration conf) {
+        this.conf = conf;
+    }
+
+    @Override
+    public ExecutorService createExecutor(ThreadFactory threadFactory, Integer concurrency, Integer maxConcurrency) {
+        final EnhancedQueueExecutor.Builder builder = new EnhancedQueueExecutor.Builder()
+                .setRegisterMBean(false)
+                .setHandoffExecutor(JBossExecutors.rejectingExecutor())
+                .setThreadFactory(JBossExecutors.resettingThreadFactory(threadFactory));
+        final int cpus = ProcessorInfo.availableProcessors();
+        // run time config variables
+        builder.setCorePoolSize(concurrency);
+        builder.setMaximumPoolSize(maxConcurrency != null ? maxConcurrency : Math.max(8 * cpus, 200));
+
+        if (conf != null) {
+            if (conf.queueSize.isPresent()) {
+                if (conf.queueSize.getAsInt() < 0) {
+                    builder.setMaximumQueueSize(Integer.MAX_VALUE);
+                } else {
+                    builder.setMaximumQueueSize(conf.queueSize.getAsInt());
+                }
+            }
+            builder.setGrowthResistance(conf.growthResistance);
+            builder.setKeepAliveTime(conf.keepAliveTime);
+        }
+
+        final EnhancedQueueExecutor eqe = builder.build();
+        eqe.prestartAllCoreThreads();
```

Comment:
I don't think we want this for the internal pool, as the majority of applications will never use it. If you use the approach I mentioned above you can remove this.

Reply:
@dmlloyd had suggested we should to not impact performance.

If we really don't think that's a problem I can take it out.

Reply:
I don't think it will matter much, in general perf is always significantly lower at startup as the JVM needs to warm up, I doubt the overhead of starting threads will be measurable in any meaningful way. The memory usage of the extra threads is measurable though, and there is a good chance that the vertx-internal pool will never be used by the majority of applications, as it is only used for 'async' file operations.

Reply:
There were some tests where I was configuring the thread pool core threads `quarkus.thread-pool.core-threads` and that setting was being ignored. The behaviour was configurable `quarkus.thread-pool.prefill`: https://github.com/quarkusio/quarkus/pull/7599/files